jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java

```diff
@@ -383,14 +389,18 @@ protected void updateWorldBound() {
 
     @SuppressWarnings("unchecked")
     public VertexBuffer[] getAllInstanceData() {
-        ArrayList<VertexBuffer> allData = new ArrayList();
+        return allInstanceData;
+    }
+
+    private void updateAllInstanceData() {
```

Comment:
I think the `read()` method needs to invoke `updateAllInstanceData()` in order to ensure that `allInstanceData` gets initialized.

Reply:
Done.